### PR TITLE
MWPW-144549 CTA alignment for Text, Icon, and Media Blocks

### PR DESCRIPTION
**Pull request by meganthecoder**
* Align CTAs of Text, Icon, and Media blocks in "-up" sections

Resolves: [MWPW-144549](https://jira.corp.adobe.com/browse/MWPW-144549)

Test URLs:
Before: https://main--milo--adobecom.hlx.page/drafts/methomas/text-alignment
After: https://methomas-text-alignment--milo--adobecom.hlx.page/drafts/methomas/text-alignment?martech=off

**Bacom:**
- Before: https://main--bacom--adobecom.hlx.live/products/experience-manager/assets/pricing
- After: https://main--bacom--adobecom.hlx.live/products/experience-manager/assets/pricing?milolibs=methomas-text-alignment
- Before: https://main--bacom--adobecom.hlx.live/
- After: https://main--bacom--adobecom.hlx.live/?milolibs=methomas-text-alignment

**Kitchen sink:**
- Text: https://methomas-text-alignment--milo--adobecom.hlx.page/docs/library/kitchen-sink/text
- Icon: https://methomas-text-alignment--milo--adobecom.hlx.page/docs/library/kitchen-sink/icon-block
- Media: https://methomas-text-alignment--milo--adobecom.hlx.page/docs/library/kitchen-sink/media

**Edge cases and regression:**
- https://methomas-text-alignment--milo--adobecom.hlx.page/drafts/methomas/text-alignment-all

UAR:
- Before: https://main--cc--adobecom.hlx.live/cc-shared/fragments/uar/marquee/marquee-plan/cci-cct
- After: https://main--cc--adobecom.hlx.live/cc-shared/fragments/uar/marquee/marquee-plan/cci-cct?milolibs=methomas-text-alignment

### CODE REVIEW

File: libs/blocks/text/text.css [hunk 1]
```diff
@@ -162,7 +167,8 @@
   justify-content: start;
 }
 
-.text-block.icon-inline .foreground [class^="body-"] {
+.text-block.icon-inline .foreground [class^="body-"],
+.text-block.icon-inline .foreground .cta-container {
```
Comment (meganthecoder) [inline libs/blocks/text/text.css:171]: This is the only update from the previous PR https://github.com/adobecom/milo/pull/2098

**Comment by meganthecoder** [comment]
@JasonHowellSlavin @Ruchika4 @ryanmparrish @elan-tbx 

I'd appreciate your re-review on this. The only new code is in text.css, lines 170-171.

You've previously reviewed this PR (here: https://github.com/adobecom/milo/pull/2098) but there was an issue found during stage testing (here: https://github.com/adobecom/milo/pull/2203#issuecomment-2077687843). It was found in the UAR quiz and applies to all Text block inline-icon variants. 

I've added that variant to the [text-alignment-all](https://methomas-text-alignment--milo--adobecom.hlx.page/drafts/methomas/text-alignment-all) page and you can also see it in the fragment from the UAR quiz on CC: https://main--cc--adobecom.hlx.live/cc-shared/fragments/uar/marquee/marquee-plan/cci-cct?milolibs=methomas-text-alignment

Thanks!

CC @JackySun9 @Brandon32

**Review by JackySun9 — APPROVED** [review]

**Review by ryanmparrish — APPROVED** [review]

File: test/blocks/media/media.test.js [hunk 2]
```diff
@@ -28,6 +28,10 @@ describe('media', () => {
       const blueButton = actionArea.querySelector('.con-button.blue');
       expect(blueButton).to.exist;
     });
+    it('has a cta container', () => {
+      const ctaArea = medias[0].querySelector('.cta-container .action-area');
+      expect(ctaArea).to.exist;
```
Comment (robert-bogos) [inline test/blocks/media/media.test.js:33]: Nit: We can turn this into a one liner by not caching ctaArea.
You can just do: `expect(medias[0].querySelector('.cta-container .action-area')).to.exist;`

File: test/blocks/media/media.test.js [hunk 3]
```diff
@@ -54,6 +58,10 @@ describe('media', () => {
       const links = medias[4].querySelectorAll('h3.heading-xs ~ p.subcopy-link > a');
       expect(links.length).to.greaterThanOrEqual(2);
     });
+    it('does not have cta container around mid-body action area', () => {
+      const actionArea = medias[4].querySelector('.action-area');
+      expect(actionArea.parentElement.className.includes('cta-container')).to.be.false;
```
Comment (robert-bogos) [inline test/blocks/media/media.test.js:63]: Same here. No need for caching.
```suggestion
      expect(medias[4].querySelector('.action-area').parentElement.className.includes('cta-container')).to.be.false;
```

File: test/blocks/text/text.test.js [hunk 4]
```diff
@@ -62,11 +62,20 @@ describe('text block', () => {
...
   describe('two content rows', () => {
     it('has viewport classes', () => {
       const mobileEl = document.querySelector('.text-block .mobile-up');
       expect(mobileEl).to.exist;
     });
   });
+  describe('cta container', () => {
+    it('is added around the last action area', () => {
+      const actionArea = document.querySelector('#has-container');
+      expect(actionArea.parentElement.className.includes('cta-container')).to.be.true;
```
Comment (robert-bogos) [inline test/blocks/text/text.test.js:74]: Same here. No need for caching
```suggestion
      expect(document.querySelector('#has-container').parentElement.className.includes('cta-container')).to.be.true;
```

File: test/blocks/text/text.test.js [hunk 5]
```diff
@@ -62,11 +62,20 @@ describe('text block', () => {
...
+    });
+    it('is not added around action areas that are not last', () => {
+      const actionArea = document.querySelector('#no-container');
+      expect(actionArea.parentElement.className.includes('cta-container')).to.be.false;
```
Comment (robert-bogos) [inline test/blocks/text/text.test.js:78]: Same here. No need for caching
```suggestion
      expect(document.querySelector('#no-container').parentElement.className.includes('cta-container')).to.be.false;
```

**Review by robert-bogos — APPROVED** [review]
Looks good 👍 I just added a few possible improvements

File: test/blocks/media/media.test.js (hunk 2 above)
Comment (meganthecoder) [reply test/blocks/media/media.test.js:33]: In general I agree but I think it improves readability of the tests so I'll keep them as is.

**Review by elan-tbx — APPROVED** [review]

**Pull request merged**